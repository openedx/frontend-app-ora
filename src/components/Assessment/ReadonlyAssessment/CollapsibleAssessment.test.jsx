import { shallow } from '@edx/react-unit-test-utils';

import CollapsibleAssessment from './CollapsibleAssessment';

describe('<CollapsibleAssessment />', () => {
  let props = {
    children: <div>Children</div>,
    stepScore: {
      earned: 5,
      possible: 10,
    },
    stepLabel: 'Step Label',
    defaultOpen: true,
  };

  it('renders the component', () => {
    const wrapper = shallow(<CollapsibleAssessment {...props} />);
    expect(wrapper.snapshot).toMatchSnapshot();

    expect(wrapper.instance.findByType('Collapsible')[0].props.open).toBe(true);
  });

  it('renders without props', () => {
    const wrapper = shallow(<CollapsibleAssessment children={props.children} />);
    expect(wrapper.snapshot).toMatchSnapshot();

    expect(wrapper.instance.findByType('Collapsible')[0].props.open).toBe(false);
  });
});