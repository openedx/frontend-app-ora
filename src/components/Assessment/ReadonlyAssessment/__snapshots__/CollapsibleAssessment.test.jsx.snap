// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CollapsibleAssessment /> renders the component 1`] = `
<Collapsible
  onToggle={[Function]}
  open={true}
  title={
    <h3>
      Step Label grade:
      5 / 10
    </h3>
  }
>
  <div>
    Children
  </div>
</Collapsible>
`;

exports[`<CollapsibleAssessment /> renders without props 1`] = `
<Collapsible
  onToggle={[Function]}
  open={false}
  title={
    <h3>
      Submitted assessment
    </h3>
  }
>
  <div>
    Children
  </div>
</Collapsible>
`;
